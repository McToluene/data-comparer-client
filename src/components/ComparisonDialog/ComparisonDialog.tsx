import React from 'react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Grid,
  Paper,
  Typography,
} from '@mui/material';
import { Company } from '../../AuthContext';

type CompanyComparisonProps = {
  companies: Company[];
  open: boolean;
  handleClose: () => void;
};

function ComparisonDialog({ companies, open, handleClose }: CompanyComparisonProps) {
  return (
    <Dialog open={open} onClose={handleClose} fullWidth maxWidth='lg'>
      <DialogTitle>Company Comparison</DialogTitle>
      <DialogContent>
        <Grid container spacing={3}>
          {companies.map((company, index) => (
            <Grid item xs={6} key={index}>
              <Paper style={{ padding: '0.7em' }} elevation={1}>
                <Typography variant='h6' gutterBottom>
                  Company: {company.name}
                </Typography>
                <Typography variant='body1'>Users: {company.users}</Typography>
                <Typography variant='body1'>Products: {company.products}</Typography>
                <Typography variant='body1'>Percentage: {company.percentage}%</Typography>

                {company.imageURL ? (
                  <img
                    src={company.imageURL}
                    alt={company.name}
                    style={{ width: '150px', height: '100px' }}
                  />
                ) : null}
              </Paper>
            </Grid>
          ))}
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color='primary'>
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
}

export default ComparisonDialog;
