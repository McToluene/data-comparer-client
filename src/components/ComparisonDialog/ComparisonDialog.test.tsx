import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For additional matchers like toBeInTheDocument
import ComparisonDialog from './ComparisonDialog';

// Mock the handleClose function
const handleClose = jest.fn();

const companies = [
  {
    id: 1,
    name: 'Company A',
    users: 100,
    products: 50,
    percentage: 75,
    imageURL: 'http://example.com/image1.jpg',
  },
  {
    id: 2,
    name: 'Company B',
    users: 200,
    products: 75,
    percentage: 90,
    imageURL: 'http://example.com/image2.jpg',
  },
];

describe('ComparisonDialog', () => {
  it('renders with companies and matches snapshot', () => {
    const { asFragment } = render(
      <ComparisonDialog companies={companies} open={true} handleClose={handleClose} />
    );

    // Check if the component matches the snapshot
    expect(asFragment()).toMatchSnapshot();

    // Check if the dialog content is rendered
    expect(screen.getByText('Company Comparison')).toBeInTheDocument();

    // Check if company information is displayed
    expect(screen.getByText('Company: Company A')).toBeInTheDocument();
    expect(screen.getByText('Company: Company B')).toBeInTheDocument();
  });

  it('renders without companies', () => {
    render(<ComparisonDialog companies={[]} open={true} handleClose={handleClose} />);

    // Check if a message for no companies is displayed
    expect(screen.getByText('Company Comparison')).toBeInTheDocument();
  });

  it('calls handleClose when close button is clicked', () => {
    render(<ComparisonDialog companies={companies} open={true} handleClose={handleClose} />);

    const closeButton = screen.getByText('Close');
    closeButton.click();

    // Check if handleClose was called
    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
