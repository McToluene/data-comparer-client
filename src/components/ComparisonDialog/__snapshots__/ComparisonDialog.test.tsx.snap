// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonDialog renders with companies and matches snapshot 1`] = `<DocumentFragment />`;
